src/miral/application_selector.cpp

```diff
@@ -250,7 +270,9 @@ auto ApplicationSelector::advance(bool reverse, bool within_app) -> Window
     else
         tools.swap_tree_order(next_window.value(), selected);
 
+    auto next_state_to_preserve =  tools.info_for(next_window.value()).state();
     tools.select_active_window(next_window.value());
+    restore_state = next_state_to_preserve;
```

Comment:
It's not clear to me why `next_state_to_preserve` exists? Why don't we immediately save it to `restore_state`?

This makes the code smell like there's spooky non-local state things happening here.

Reply:
When we select the active window, the current state of the window _might_ change state if it was previously minimized. It will go from minimized to restored, but we want minimized to be saved.

On top of that, when we focus a new window, we restore the state of the previously-selected window using `restore_state`, so we don't want to override it before we call `select_active_window`